### PR TITLE
Allow `Optimizer`s as input to `VQEProgram`

### LINKED ISSUE #289: `VQEProgram` to support optimizers classes for optimizers instead of dictionaries
### What is the expected behavior?

Since Terra 0.18 optimizers are serializable and supported by the runtime VQE script. The VQEProgram must be updated to support them, too. Currently it stil expects the optimizer as dictionary.

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Fixes #289.

### Details and comments

Dictionaries are still supported since some tutorials first need to be updated to use optimizers.

### CODE REVIEW

**Comment by woodsp-ibm** [comment]
So this is a feature rather than any bug fix then. I see you put a high priority label on the issue which is why I am asking. As a feature this would go out in the next release unless somehow its creating critical problems/issues out there now. I assume you will be updating the optimization repo equivalently. If we need these to stay the same out there so users can use the one in Nature and/or the one in Optimization the same way then I guess things need to be coordinated/sorted out, since I presently would have expected a Nature release in advance of the next Optimization one - unless we start talking bug fix releases.

File: releasenotes/notes/vqeprogram-optimizers-cd23307c10fcd15b.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+features:
```
Comment (Cryoris) [inline releasenotes/notes/vqeprogram-optimizers-cd23307c10fcd15b.yaml:2]:
```suggestion
fixes:
```

**Comment by Cryoris** [comment]
It would be better to get this out before the next feature release, right, so it seems more like a bugfix 😉 I synced with @pbark and looks like we can do a minor release soon.

**Review by pbark — APPROVED** [review]
This LGTM!

**Review by mrossinek — APPROVED** [review]

**Review by woodsp-ibm — APPROVED** [review]

**Pull request merged**